### PR TITLE
simplify awxkit dependencies

### PR DESCRIPTION
**Pull request by ryanpetrello**
- remove flake8 as an install requirements (it's only used for tests)
- vendor toposort, which is Apache 2.0 licensed (and very small)
- change websocket-client to a setuptools optional dependency, which you
  can install via:

  `pip install "./awxkit[websockets]"`

- add `jq` and `tabulate` under an additional optional setuptools
  dependency:

  `pip install "./awxkit[formatting]"`

### CODE REVIEW

File: awxkit/requirements.txt [hunk 1]
```diff
@@ -1,8 +1,4 @@
 PyYAML>=5.1
-cryptography
```
Comment (kdelee) [inline awxkit/requirements.txt:2]: might want to confirm awxkit tests don't need this

File: awxkit/awxkit/utils/toposort.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+class CircularDependencyError(ValueError):
+    def __init__(self, data):
+        # Sort the data just to make the output consistent, for use in
+        #  error messages.  That's convenient for doctests.
+        s = 'Circular dependencies exist among these items: {{{}}}'.format(', '.join('{!r}:{!r}'.format(key, value) for key, value in sorted(data.items())))
+        super(CircularDependencyError, self).__init__(s)
+        self.data = data
+
+
+def toposort(data):
```
Comment (ryanpetrello) [inline awxkit/awxkit/utils/toposort.py:50]: This is such a tiny dependency that it's not worth carrying externally.

File: awxkit/awxkit/ws.py [hunk 3]
```diff
@@ -53,6 +51,9 @@ class WSClient(object):
     # Subscription group types
 
     def __init__(self, token=None, hostname='', port=443, secure=True, session_id=None, csrftoken=None):
+        # delay this import, because this is an optional dependency
```
Comment (ryanpetrello) [inline awxkit/awxkit/ws.py:54]: the CLI doesn't use and doesn't need this

File: awxkit/requirements.txt [hunk 4]
```diff
@@ -1,8 +1,4 @@
...
-flake8
 python-dateutil
 requests
 termcolor
```
Comment (ryanpetrello) [inline awxkit/requirements.txt:4]: @shanemcd I think `termcolor` is _maybe_ the only thing on this list that might require some work on our end

File: awxkit/requirements.txt (hunk 1 above)
Comment (ryanpetrello) [reply awxkit/requirements.txt:2]: doesn't look to me like they do

**Review by matburt — APPROVED** [review]

**Pull request merged**